k8s/provider_test.go

```diff
@@ -351,3 +352,108 @@ func marshalPB(obj interface{}) ([]byte, error) {
 	copy(d[len(magicBytes):], body)
 	return d, nil
 }
+
+func TestGenerateServicesWithPropperHealthCheck(t *testing.T) {
+	setEnv(t, "testdata/port_definitions_probe_and_service_only.json")
+	defer unsetEnv(t)
+
+	pod := testPod()
+	path := "/status/ping"
+	pod.Spec.Containers[0].LivenessProbe = &corev1.Probe{
+		Handler: &corev1.Handler{
+			HttpGet: &corev1.HTTPGetAction{
+				Path: &path,
+			},
+		},
+	}
+
+	services, err := generateServices("serviceName", pod, nil)
+	require.NoError(t, err)
+
+	expectedServices := []consul.ServiceInstance{
+		{
+			Check: &consul.Check{
+				Address: "http://192.0.2.2:0/status/ping",
+				Type: "HTTP_GET",
+			},
+		},
+	}
+
+	assert.Len(t, services, 1)
+	assert.Equal(t, expectedServices[0].Check, services[0].Check)
+}
+
+func TestCheckOneServiceFromWithProperTags(t *testing.T) {
+	setEnv(t, "testdata/port_definitions_probe_and_service_only.json")
+	defer unsetEnv(t)
+
+	pod := testPod()
+	tags := []string{"a", "b", "c"}
+	services, err := generateServices("serviceName", pod, tags)
+	require.NoError(t, err)
+
+	expectedServices := []consul.ServiceInstance{
+		{
+			Tags: []string{
+				"a", "b", "c",
+			},
+		},
+	}
+
+	assert.Len(t, services, 1)
+	assert.Equal(t, expectedServices[0].Tags, services[0].Tags)
+}
+
+func TestShouldCheckMultipleServicesWithGlobalTagsCombinedAndWithPortSpecificTags(t *testing.T) {
+	setEnv(t, "testdata/port_definitions_service_probe_secured.json")
+	defer unsetEnv(t)
+	pod := testPod()
+	tags := []string{"a", "b", "c"}
+	services, err := generateServices("serviceName", pod, tags)
```

Comment:
Separate this //when block from the //given part, that helps readability significantly. Also, move `expectedServices` here.